### ISSUE TITLE
How to deserialize TomlNode?

### ISSUE THREAD

**Issue by Xirado** [orchestr7/ktoml#280]
I need to dynamically traverse a toml file since i cannot serialize the whole thing, but have not found a way to deserialize a TomlNode into a kotlinx Serializable class. Is it possible? If not, i think adding this functionality would be beneficial

**Comment by orchestr7** [comment]
> I need to dynamically traverse a toml file since i cannot deserialize the whole thing, but have not found a way to deserialize a TomlNode into a kotlinx Serializable class. Is it possible? If not, i think adding this functionality would be beneficial

Hey! Yes, we have a partial serialization. It is very easy

**Comment by orchestr7** [comment]
Check this part in the readme![image](https://github.com/user-attachments/assets/e8009f50-af36-46d6-92ce-64d00906c22f)

**Comment by Xirado** [comment]
> I think this is your scenario. Check this part in the readme![image](https://private-user-images.githubusercontent.com/58667063/370882697-e8009f50-af36-46d6-92ce-64d00906c22f.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyOTYyNzcsIm5iZiI6MTcyNzI5NTk3NywicGF0aCI6Ii81ODY2NzA2My8zNzA4ODI2OTctZTgwMDlmNTAtYWYzNi00NmQ2LTkyY2UtNjRkMDA5MDZjMjJmLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyNVQyMDI2MTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNDk2MDEzMjFhZTNmYzU3NDI2Mzk3MGU3MzZmNGE0OGI5MTA0YjZlMjhkZDhjY2YwZGIzZjU4Yzk2MmZkMzUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xF58sKIYAW-ZwXmo9qrOQiMyvmL5vw6eq6-V_AQ0fSk)

Yes, i read about this, but shouldn't there be a more efficient way? Every call will cause the whole file to be parsed again, which is very inefficient, `TomlTable` already has all the information, so i think there should be a way to serialize that aswell since it only requires reading the file once

**Comment by akuleshov7** [comment]
> Yes, i read about this, but shouldn't there be a more efficient way? Every call will cause the whole file to be parsed again, which is very inefficient, `TomlTable` already has all the information, so i think there should be a way to deserialize that aswell since it only requires reading the file once

Hm, may be I am getting this wrong, but if your file is changing dynamically and you re-read it again and again, then there anyway will be same parsing. 

May be you can give some small example, so we can implement it?

**Comment by Xirado** [comment]
> > Yes, i read about this, but shouldn't there be a more efficient way? Every call will cause the whole file to be parsed again, which is very inefficient, `TomlTable` already has all the information, so i think there should be a way to deserialize that aswell since it only requires reading the file once
> 
> Hm, may be I am getting this wrong, but if your file is changing dynamically and you re-read it again and again, then there anyway will be same parsing.
> 
> May be you can give some small example of the input and API, so we can implement it?

The file isn't changing, but i need to read the file as such

```toml
[a]
foo = "foo"
bar = "bar"

[b.a]
foo = "foo"
bar = "bar"

[b.b]
foo = "foo"
bar = "bar"

[x.y.z]
foo = "foo"
bar = "bar"
```

and then decode the file as such that i get `Map<String, Foobar>` where the String represents the path, so e.g. `x.y.z`
